libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -77,6 +77,11 @@ void AP_Motors::armed(bool arm)
 
 void AP_Motors::set_desired_spool_state(DesiredSpoolState spool)
 {
+    // check if spoolup has been blocked
+    if (_spoolup_block && (spool == DesiredSpoolState::THROTTLE_UNLIMITED)) {
```

Comment:
Rather than blocking the change of desired spool state could we accept the desired spool state but never change the actual. So copter could ask for UNLIMITED bit it would not move out of `GROUND_IDLE` until it get the all clear. 

Basically it would block this:
https://github.com/ArduPilot/ardupilot/blob/d31f6c2f57e707db8bea526735d013d2de1adaad/libraries/AP_Motors/AP_MotorsMulticopter.cpp#L610-L617

Reply:
@IamPete1, txs for the review.

The danger of moving the block lower down is it means that each motor backend would need to be sure to implement the block correctly.

Reply:
There are really only two backends, multi and heli, this already doesn't work for heli. Everything else inherits.

Reply:
I think we agree with @IamPete1 that instead of rejecting the set of desired spool state we should block the actual spool state from progressing.

Reply:
Ok, a spool up block makes sense. 

However this could be dangerous if someone accidently pulls the interlock in flight and releases it. If one of the motors doesn't start properly then the aircraft would stay shut down and an X8 would then crash unnecessarily. So we would need to make sure the check turns the block off if the aircraft is flying.

Reply:
This should definatly be in the code that Peter points out rather than preventing the desired spool state from being set in the first place.